### PR TITLE
checkPassword added and user tests updated

### PR DESCRIPTION
**Pull request by StefanoPitto**

### CODE REVIEW

**Review by HaseebUllahAbbasi — COMMENTED** [review]
the integration test also pass and changes are according to the requirements
 but @theblockstalk @Rebal17  might approve this

File: test-integration/user.test.ts [hunk 1]
```diff
@@ -199,6 +199,21 @@ describe('User class', () => {
...
+            await user.login(await user.getUsername(), password);
+
+            await expect(user.checkPassword(password)).resolves.toBeTruthy();
+            await expect(user.checkPassword('Testing123!@')).rejects.toThrowError(SdkErrors.PasswordInValid);
+            await expect(user.checkPassword('password')).rejects.toThrowError(SdkErrors.PasswordFormatInvalid);
+
+            await user.logout();
+        })
+    );
+
```
Comment (Rebal67) [inline test-integration/user.test.ts:216]: this is good but we also need a success case so one more test that returns true if password matches 👍

Comment (StefanoPitto) [reply test-integration/user.test.ts:216]: isn't it already on line 209?

Comment (Rebal67) [reply test-integration/user.test.ts:216]: no because this function was newly made by you so we want to test the success case

**Pull request merged**